src/app/campaign.model.ts

```diff
@@ -1,12 +1,35 @@
 /**
  * @link https://app.swaggerhub.com/apis/Noel/TBG-Campaigns/#/Campaign
+ *
+ * Represents an individual campaign to raise funds for one charity - distinct from a metacampaign
+ * which has its own model
  */
-type CampaignOrMetaCampaign = {
```

Comment:
losing the shared base type, I think explicitly repeating the properties that are used in both types is clearer than having a shared base type, since they're more different than similar.